tools/rosgraph/src/rosgraph/xmlrpc.py

```diff
@@ -76,9 +77,40 @@ def isstring(s):
     except NameError:
         return isinstance(s, str)
 
-class SilenceableXMLRPCRequestHandler(SimpleXMLRPCRequestHandler):
 
-    protocol_version = 'HTTP/1.1'
+def _support_http_1_1():
+    """Linux kernels 4.15 and older encounter performance issues with HTTP/1.1
+    This function returns true if the Linux kernel version is greater than the
+    input or if the platform is not Linux.
+    """
+    # HTTP/1.1 performance issue only detected on Linux
+    if platform.system() != 'Linux':
+        return True
+
+    release = platform.release().split('.')
+
+    major = 4
+    minor = 15
+
+    platform_major = int(release[0])
+    platform_minor = int(release[1])
+
+    if platform_major > major:
```

Comment:
Are we going for ">=4.15" as the target of this logic? Just thinking out loud, but I had a mildly difficult time getting that from the blocks below - the following might be easier to read

```
if platform_major < major:
    return False
if platform_major == major and platform_minor <= minor:
    return False
return True

```